static/js/user_topics.js

```diff
@@ -7,6 +7,13 @@ import {get_time_from_date_muted} from "./util";
 
 const muted_topics = new Map();
```

Comment:
I think we'll need a more general `user_topics` map in the future. We can rename this `muted_topics` map to `user_topics` and store (`last_updated`, `visibility_policy`) in it just the same way we are doing `date_muted` now.

Adding and updating `user_topic` data would be easy this way, but in functions like `get_muted_topics`, it won't be as efficient as we'll have to filter through all types of records.

I'd like some feedback on this thought.

Reply:
Yeah, I think renaming this to `user_topics` seems right. 

I think what we will want is for internal data structures like this to just map (stream => topic => object), rather than a timestamp at the bottom. 

I think it'll still make sense to have API functions like `user_topics.is_topic_muted(stream_id, topic)` that hide the complexity from callers where it's not relevant.